fauxfactory/factories/numbers.py

```diff
@@ -60,3 +60,29 @@ def gen_positive_integer():
     min_value = 0
 
     return gen_integer(min_value=min_value)
+
+
+def gen_octagonal(min_value=None, max_value=None):
+    """Return a random octagonal number based on the current platform.
+
+    :returns: A random hexadecimal number.
+    :rtype: str
+
+    """
+    return oct(gen_integer(
+        int(str(min_value), base=8) if min_value else min_value,
+        int(str(max_value), base=8) if max_value else max_value
+    ))
+
+
+def gen_hexadecimal(min_value=None, max_value=None):
```

Comment:
I guess same comments for the `gen_octagonal` method applies here. I think that the method can be simplified like this:

```python
return hex(gen_integer(min_value, max_value))
```